third_party/proton/csrc/lib/Profiler/RocTracer/RoctracerProfiler.cpp

```diff
@@ -364,8 +364,9 @@ void RoctracerProfiler::RoctracerProfilerPimpl::activityCallback(
                     externId, dataSet, record, isAPI, isGraph);
     // Track correlation ids from the same stream and erase those <
     // correlationId
-    correlation.corrIdToExternId.erase(record->correlation_id);
-    correlation.apiExternIds.erase(externId);
+    if (!correlation.corrIdToExternId.contain(record->correlation_id)) {
```

Comment:
If it doesn't contain the correlation id, can we just skip this record and process the next?

Reply:
I meant a bigger refactoring.

```
if correlation.corrIdToExternId.contain(record->correlation_id) {
processActivity...
}
roctracer::getNextRecord<true>(record, &record);
```

So when `corrIdToExternId` doesn't find `correlation_id` we just skip to the next record